src/test/osd/osdcap.cc

```diff
@@ -270,6 +270,11 @@ TEST(OSDCap, BasicRW) {
   ASSERT_FALSE(cap.is_capable("bar", "", 0, "foo", false, true, false, true));
   ASSERT_FALSE(cap.is_capable("bar", "", 0, "foo", true, false, true, false));
   ASSERT_FALSE(cap.is_capable("bar", "", 0, "foo", true, true, true, true));
+
+  // class whitelist flag works
+  ASSERT_TRUE(cap.is_capable("bar", "", 0, "foo", false, true, false, true, true));
+  ASSERT_TRUE(cap.is_capable("bar", "", 0, "foo", true, false, true, false, true));
+  ASSERT_TRUE(cap.is_capable("bar", "", 0, "foo", true, true, true, true, true));
```

Comment:
yeah, 'allow *' is like root.

On Tue, 28 Jun 2016, Noah Watkins wrote:

> 
> In src/test/osd/osdcap.cc:
> 
> > @@ -270,6 +270,11 @@ TEST(OSDCap, BasicRW) {
> >    ASSERT_FALSE(cap.is_capable("bar", "", 0, "foo", false, true, false, true));
> >    ASSERT_FALSE(cap.is_capable("bar", "", 0, "foo", true, false, true, false));
> >    ASSERT_FALSE(cap.is_capable("bar", "", 0, "foo", true, true, true, true));
> > +
> > +  // class whitelist flag works
> > +  ASSERT_TRUE(cap.is_capable("bar", "", 0, "foo", false, true, false, true, true));
> > +  ASSERT_TRUE(cap.is_capable("bar", "", 0, "foo", true, false, true, false, true));
> > +  ASSERT_TRUE(cap.is_capable("bar", "", 0, "foo", true, true, true, true, true));
> 
> does 'allow *' override the whitelist?
> 
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub, or mute the
> thread.[AAabh_n9-cs0YNVIW8IwkQQRxb2aCuOoks5qQW4agaJpZM4I_hOr.gif]
> 
> 
>